src/logger/src/metrics.rs

```diff
@@ -223,16 +237,19 @@ impl Metric for SharedMetric {
     fn count(&self) -> usize {
         self.0.load(Ordering::Relaxed)
     }
+}
+
+impl StoreMetric for SharedStoreMetric {
+    fn fetch(&self) -> usize {
+        self.0.load(Ordering::Relaxed)
+    }
 
-    // It is necessary to also reset the old value in order to flush later the correct one,
-    // which is `value` (see the `Serialize` implementation for SharedMetric a few lines below).
     fn store(&self, value: usize) {
         self.0.store(value, Ordering::Relaxed);
-        self.1.store(0, Ordering::Relaxed)
```

Comment:
Indeed this was racy between thread doing `store` and thread doing `serialize`, good catch!

Reply:
hmm, was this racy? I think the atomic type prevented that, right?

Reply:
makes total sense, thanks